GoogleForCreators/web-stories-wp
# Add option for choosing custom archive page

## Context



We want to allow users to use a custom static page for the /web-stories archive page. See #8812 and #4862 for full context.

## Summary



**Note:** Depends on #8958 being merged first.

This PR adds a new option to the Archive Page dropdown in the dashboard settings to allow users to choose a custom archive page.

## Relevant Technical Choices



## To-do



- Tests
- Code cleanup

In the future, it would be great to handle this scenario:

1. User has currently chosen page XYZ as the archive page
2. User goes to settings, sees link to `example.com/xyz` there
3. User changes setting to "Default"
4. Link is not updated to `example.com/web-stories/`

To fix this, the URL would need to be retrieved from the REST API instead of being passed via config.

## User-facing changes



![Screenshot 2021-09-17 at 00 58 02](https://user-images.githubusercontent.com/841956/133696011-0b49f75b-b33a-46f3-8224-46e283c9a2d3.png)

![Screenshot 2021-09-21 at 19 20 45](https://user-images.githubusercontent.com/841956/134217764-f5f88bf2-9fa1-4779-96a9-5dcd35d8cf7b.png)

## Testing Instructions






This PR can be tested by following these steps:

1. Enable archive page customizaton experiment
2. Go to story settings and choose between three archive page options (default, disabled, custom)
3. Verify that archive page URL is now the page URL, showing page contents instead of default archive
4. Visit page list table in wp-admin, see "Web Stories Archive Page" label next to the chosen custom page

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #8812
